File: debug_patterns.exs
#!/usr/bin/env elixir

# Quick debug script to check what patterns are returned
patterns = RsolvApi.Security.ASTPattern.get_patterns("javascript", :public, :standard)
pattern_ids = Enum.map(patterns, & &1.id)

IO.puts("Number of patterns: #{length(patterns)}")
IO.puts("Pattern IDs:")
IO.inspect(pattern_ids, label: "Pattern IDs")

IO.puts("\nExpected patterns from test:")
expected = [
  "js-sql-injection-concat",
  "js-sql-injection-interpolation", 
  "js-xss-innerhtml",
  "js-xss-document-write",
  "js-command-injection-exec"
]

IO.inspect(expected, label: "Expected")

IO.puts("\nMissing patterns:")
missing = expected -- pattern_ids
IO.inspect(missing, label: "Missing")

IO.puts("\nExtra patterns:")
extra = pattern_ids -- expected
IO.inspect(extra, label: "Extra")

File: debug_patterns.ts
#!/usr/bin/env bun

// Debug script to see what patterns we're getting

async function debugPatterns() {
  const response = await fetch('https://api.rsolv-staging.com/api/v1/patterns?language=javascript&format=enhanced');
  const data = await response.json();
  
  console.log("Patterns received:");
  for (const pattern of data.patterns) {
    console.log(`\n${pattern.id} (${pattern.type}):`);
    console.log("  regex_patterns:", pattern.regex_patterns);
    console.log("  ast_rules:", JSON.stringify(pattern.ast_rules, null, 2));
    console.log("  context_rules:", JSON.stringify(pattern.context_rules, null, 2));
  }
}

debugPatterns().catch(console.error);